Fix Vanilla Template v.2

MarcSkovMadsen opened the pull request:
Supersedes https://github.com/holoviz/panel/pull/1642. I had to fix some line endings and Git base branch issues.

----------

I've tested out the Vanilla Template and could see it would not work for users. 

I've listed some issues here https://github.com/holoviz/panel/issues/1641 that this PR tries to solve.

You can try out the vanilla template example by running

```python
python -m panel serve 'panel\tests\template\test_vanilla_manual.py' --dev
```

![vanilla2](https://user-images.githubusercontent.com/42288570/96376821-710c0a80-1181-11eb-9cfc-674476118ced.gif)


Comments:

- I've fixed the sidebar width at `300px` as that in my experience is a good width. It's also the width used by Streamlit and I've never heard users propose any change to that. Maybe the width should be user customizable such that you can specify a string like `250px` or `16.67vw`?
- I've added a little bit more margin/ padding (~`10px`) in different places to make things look a little less busy. Maybe that should be user customizable as well.
- I've set `overflow-x: hidden` on both sidebar and main area. In my experience that is a very, very good solution. But if some people have very wide plots or images they would like to show it will not work. But in my experience there should be as few scrollbars as possible. There is no horizontal scrollbar in the Streamlit template and I've never heard users complain.
- I've given the sidebar just a little bit of gray to more clearly separate it from the rest.

FYI @nghenzi and @philippjfr . This might have the implications for the other templates and themes like the React Template.

MarcSkovMadsen commented:
I believe this template will be useable on mobile to some extent. The only thing that would be nice would be if it was possible to assign a size grid of 12 columns like the Bootstrap or Material Grid such that the main items would be layed out differently on a widescreen and on mobile.

![vanilla_mobile](https://user-images.githubusercontent.com/42288570/96377085-415e0200-1183-11eb-8938-89bdd1d67ec5.gif)

MarcSkovMadsen commented:
I've also fixed the issue with the 300px margin when there is no sidebar.

![image](https://user-images.githubusercontent.com/42288570/96377206-58e9ba80-1184-11eb-9983-1f3c95d8dee0.png)

https://github.com/holoviz/panel/issues/1644

MarcSkovMadsen commented:
I've added a new parameter called `main_max_width` to the template. The reason is that one of the basic things in a lot of web pages is that they center the main content and give it a max size in order to make sure it's not too wide to read/ work with.

I've set the `main_max_width` to '' (empty string) by default. And it will not be applied.

The screenshots below show examples with main_max_width of `600px`.

![image](https://user-images.githubusercontent.com/42288570/96377618-0fe73580-1187-11eb-8304-9bee891b448f.png)

![image](https://user-images.githubusercontent.com/42288570/96377628-183f7080-1187-11eb-9563-57600dc3da62.png)

MarcSkovMadsen commented:
FYI @philippjfr . 

I've "fixed" the issues I have seen and tried to improve it based on the experiences I have had with the awesome-panel.org template and site for almost a year.

I don't plan to do further improvements except based on your review.

philippjfr commented:
Really appreciate your work here. I expect a rapid series of 0.10.x releases just for fixes like this at this point. Will review tomorrow and then I'm pretty much ready for the 0.10.0 release by about Tuesday. The blog post is finally written and I should have some time to address final issues tomorrow.

MarcSkovMadsen commented:
I've done some further improvements. I know from other communities that being able to use custom favicons and support search engine optimization is badly requested.

So I've added support for favicon, meta and base data. I also changed the code a bit so the title is also shown in the browser tab instead of always 'Panel Application'.

![image](https://user-images.githubusercontent.com/42288570/96404796-15bb3600-11dc-11eb-8edc-d351795bb1c3.png)

![image](https://user-images.githubusercontent.com/42288570/96404840-34b9c800-11dc-11eb-93be-802d41f4167f.png)

MarcSkovMadsen commented on panel/template/base.py:
```diff
@@ -37,6 +37,9 @@
 )
 
+# Should be served from somewhere else!
+FAVICON_URL = "https://raw.githubusercontent.com/MarcSkovMadsen/awesome-panel/2781d86d4ed141889d633748879a120d7d8e777a/assets/images/favicon.ico"
```
You should serve this from somewhere else @philippjfr

MarcSkovMadsen commented on panel/template/base.py:
```diff
@@ -393,6 +430,7 @@ def _css_files(self):
         return css_files
 
     def _init_doc(self, doc=None, comm=None, title=None, notebook=False, location=True):
+        title = title or self.title
```
Import to be able to show user specified title in browser tab.

philippjfr commented:
First comment, would you mind removing unrelated changes to the CHANGELOG, alert and .git files?

MarcSkovMadsen commented:
Hi @philippjfr . As stated above Please review (and merge) https://github.com/holoviz/panel/pull/1646 first. Then those CHANGELOG, alert and .git files changes should go away AND THE CHANGES SHOULD BE CLEAR AS THERE IS NO WINDOWS/ LINUX line endings discrepancies.

Let me know if this is not possible.

philippjfr commented:
Going to merge this and follow up in another PR.
